doc/metadata_routing.rst

```diff
@@ -0,0 +1,209 @@
+
+.. _metadata_routing:
+
+.. TODO: update doc/conftest.py once document is updated and examples run.
+
+Metadata Routing
+================
+
+This guide demonstrates how metadata such as ``sample_weight`` can be routed
+and passed along to estimators, scorers, and CV splitters through
+meta-estimators such as ``Pipeline`` and ``GridSearchCV``. In order to pass
+metadata to a method such as ``fit`` or ``score``, the object accepting the
+metadata, must *request* it. For estimators and splitters this is done via
+``set_*_request`` methods, e.g. ``set_fit_request(...)``, and for scorers this
+is done via ``set_score_request`` method. For grouped splitters such as
+``GroupKFold`` a ``groups`` parameter is requested by default. This is best
+demonstrated by the following examples.
+
+Usage Examples
+**************
+Here we present a few examples to show different common use-cases. The examples
+in this section require the following imports and data::
+
+  >>> import numpy as np
+  >>> from sklearn.metrics import make_scorer, accuracy_score
+  >>> from sklearn.linear_model import LogisticRegressionCV
+  >>> from sklearn.linear_model import LogisticRegression
+  >>> from sklearn.model_selection import cross_validate
+  >>> from sklearn.model_selection import GridSearchCV
+  >>> from sklearn.model_selection import GroupKFold
+  >>> from sklearn.feature_selection import SelectKBest
+  >>> from sklearn.utils.metadata_requests import RequestType
+  >>> from sklearn.pipeline import make_pipeline
+  >>> n_samples, n_features = 100, 4
+  >>> X = np.random.rand(n_samples, n_features)
+  >>> y = np.random.randint(0, 2, size=n_samples)
+  >>> my_groups = np.random.randint(0, 10, size=n_samples)
+  >>> my_weights = np.random.rand(n_samples)
+  >>> my_other_weights = np.random.rand(n_samples)
+
+Weighted scoring and fitting
+----------------------------
+
+Here ``GroupKFold`` requests ``groups`` by default. However, we need to
+explicitly request weights in ``make_scorer`` and for ``LogisticRegressionCV``.
+Both of these *consumers* know how to use metadata called ``"sample_weight"``::
+
+  >>> weighted_acc = make_scorer(accuracy_score).set_score_request(
+  ...     sample_weight=True
+  ... )
+  >>> lr = LogisticRegressionCV(
+  ...     cv=GroupKFold(), scoring=weighted_acc,
+  ... ).set_fit_request(sample_weight=True)
+  >>> cv_results = cross_validate(
+  ...     lr,
+  ...     X,
+  ...     y,
+  ...     cv=GroupKFold(),
+  ...     props={"sample_weight": my_weights, "groups": my_groups},
+  ...     scoring=weighted_acc,
+  ... )
+
+Note that in this example, ``my_weights`` is passed to both the scorer and
+``~linear_model.LogisticRegressionCV``.
+
+Error handling: if ``props={'sample_weigh': my_weights, ...}`` were passed
+(note the typo), cross_validate would raise an error, since 'sample_weigh' was
```

Comment:
```suggestion
Error handling: if ``props={"sample_weigh": my_weights, ...}`` were passed
(note the typo), cross_validate would raise an error, since "sample_weigh" was
```